sdk/core/azure-json-reflect/src/main/java/com/azure/json/reflect/Version.java

```diff
@@ -0,0 +1,54 @@
+package com.azure.json.reflect;
+
+public class Version implements Comparable<Version> {
```

Comment:
We have a strong desire to not have API exposed publicly for things that we don't particularly want end users to see or use. I suspect this class falls into that category, and if so, we should consider the right path forward to hide this type. We have two first choices:

1. Make the class be package-private. This means it is not visible at all outside of the package, which is obviously perfect, if the Version class is only used within this package.
2. We have a pattern of having an `implementation` package for code that is not intended for end users. We could do this here - which would result in a package `com.azure.json.reflect.implementation`

This comment applies to all code below - if it in't intended for public consumption, we should refactor to reduce its visibility.